lib/Ceilometer/Types/Instance.hs

```diff
@@ -60,6 +62,36 @@ $(declarePF    "Instance"
 
 -- BOILERPLATE GALORE
 
+isInstanceFlavorBillable :: PDInstanceFlavor -> Bool
+isInstanceVCPUBillable   :: PDInstanceVCPU   -> Bool
+isInstanceRAMBillable    :: PDInstanceRAM    -> Bool
+isInstanceDiskBillable   :: PDInstanceDisk   -> Bool
+isInstanceFlavorBillable (PDInstanceFlavor status _) = isInstanceStatusBillable status
+isInstanceVCPUBillable   (PDInstanceVCPU   status _) = isInstanceStatusBillable status
+isInstanceRAMBillable    (PDInstanceRAM    status _) = isInstanceStatusBillable status
+isInstanceDiskBillable   (PDInstanceDisk   status _) = isInstanceStatusBillable status
+
+isInstanceStatusBillable :: PFInstanceStatus -> Bool
+isInstanceStatusBillable InstanceError            = False
+isInstanceStatusBillable InstanceActive           = True
+isInstanceStatusBillable InstanceShutoff          = False
+isInstanceStatusBillable InstanceBuild            = False
+isInstanceStatusBillable InstanceRebuild          = False
+isInstanceStatusBillable InstanceDeleted          = False
+isInstanceStatusBillable InstanceSoftDeleted      = False
+isInstanceStatusBillable InstanceShelved          = False
+isInstanceStatusBillable InstanceShelvedOffloaded = False
+isInstanceStatusBillable InstanceReboot           = True
+isInstanceStatusBillable InstanceHardReboot       = True
+isInstanceStatusBillable InstancePassword         = True
+isInstanceStatusBillable InstanceResize           = True
+isInstanceStatusBillable InstanceVerifyResize     = True
+isInstanceStatusBillable InstanceRevertResize     = True
+isInstanceStatusBillable InstancePaused           = False
+isInstanceStatusBillable InstanceSuspended        = False
+isInstanceStatusBillable InstanceRescue           = False
+isInstanceStatusBillable InstanceMigrating        = False
+
 data PDInstanceVCPU = PDInstanceVCPU PFInstanceStatus PFValue32
```

Comment:
can we not have Java names? billableFlavor, billableStatus, etc. is fine

also, I thought we're moving this mapping to borel?